### ISSUE TITLE
Client onConnected / onDisconnected callbacks

### ISSUE THREAD

**Issue by jamesadevine** [microsoft/jacdac#346]
Would be nice to have a callback to know when a client has been connected/disconnected from a server. It would also be good to have an api like: `isConnected` for polling code too!

**Comment by pelikhan** [comment]
Blocked by https://github.com/microsoft/jacdac/issues/351

**Comment by pelikhan** [comment]
Upgrade to 0.4.47

![image](https://user-images.githubusercontent.com/4175913/115814381-c4b61580-a3a9-11eb-8b4a-e786e574be9e.png)
